### PR TITLE
core/config/v2/docs: document secrets

### PR DESCRIPTION
**Pull request by jmank88**
https://app.shortcut.com/chainlinklabs/story/54383/document-secrets-toml

This PR documents the `secrets.toml` fields and their environment variable alternatives by adding a second generated file `docs/SECRETS.md` alongside `docs/CONFIG.md`. Two files is simpler for a variety of reasons, and we can still link between them. We could try to combine them if there is a compelling reason, but it would be a much larger refactor.

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -148,7 +148,7 @@ test_chaos: test_need_operator_assets ## Run core node chaos tests.
 
 .PHONY: config-docs
 config-docs: ## Generate core node configuration documentation
-	go run ./core/config/v2/docs/cmd/generate/main.go > ./docs/CONFIG.md
+	go run ./core/config/v2/docs/cmd/generate/main.go -o ./docs/
```
Comment (jmank88) [inline GNUmakefile:151]: Target an output directory now since there are two files.

File: core/config/v2/docs/docs.go [hunk 2]
```diff
@@ -34,18 +36,33 @@ var (
...
+}
+
+// GenerateSecrets returns MarkDown documentation generated from secrets.toml.
+func GenerateSecrets() (string, error) {
+	return generateDocs(secretsTOML, `[//]: # (Documentation generated from docs/secrets.toml - DO NOT EDIT.)
+
+This document describes the TOML secrets format.
+
+See also [CONFIG.md](config.md)
+`)
```
Comment (jmank88) [inline core/config/v2/docs/docs.go:56]: We should say more here from a high level, but perhaps out of scope ATM.

File: core/config/v2/docs/secrets.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# BackupURL is where the automatic database backup will pull from, rather than the main DATABASE_URL. It is recommended
+# to set this value to a read replica if you have one to avoid excessive load on the main database.
+#
+# Environment variable alternative:
+# `CL_DATABASE_BACKUP_URL=postgresql://user:pass@localhost:5432/backupdbname?sslmode=disable`
+BackupURL = "postgresql://user:pass@localhost:5432/backupdbname?sslmode=disable" # Example
+# AllowSimplePasswords skips the password complexity check.
+#
+# TODO need an env var?
+AllowSimplePasswords = false # Default
```
Comment (jmank88) [inline core/config/v2/docs/secrets.toml:17]: If someone is using env vars for their secrets, is it a burden to have to create a `secrets.toml` just to use this field?

File: core/config/v2/docs/secrets.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# [Running a Chainlink Node](https://docs.chain.link/docs/running-a-chainlink-node/#set-the-remote-database_url-config) for an example.
+#
+# Environment variable alternative:
+# `CL_DATABASE_URL=postgresql://user:pass@localhost:5432/dbname?sslmode=disable`
+URL = "postgresql://user:pass@localhost:5432/dbname?sslmode=disable" # Example
+# BackupURL is where the automatic database backup will pull from, rather than the main DATABASE_URL. It is recommended
+# to set this value to a read replica if you have one to avoid excessive load on the main database.
+#
+# Environment variable alternative:
+# `CL_DATABASE_BACKUP_URL=postgresql://user:pass@localhost:5432/backupdbname?sslmode=disable`
```
Comment (jmank88) [inline core/config/v2/docs/secrets.toml:12]: Maybe the values don't add much here...
```suggestion
# Environment variable: `CL_DATABASE_BACKUP_URL`
```

File: core/config/v2/docs/secrets.toml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+# URL is the PostgreSQL URI to connect to your database. Chainlink nodes require Postgres versions >= 11. See
+# [Running a Chainlink Node](https://docs.chain.link/docs/running-a-chainlink-node/#set-the-remote-database_url-config) for an example.
+#
+# Environment variable: `CL_DATABASE_URL`
+URL = "postgresql://user:pass@localhost:5432/dbname?sslmode=disable" # Example
+# BackupURL is where the automatic database backup will pull from, rather than the main DATABASE_URL. It is recommended
+# to set this value to a read replica if you have one to avoid excessive load on the main database.
+#
+# Environment variable: `CL_DATABASE_BACKUP_URL`
+BackupURL = "postgresql://user:pass@localhost:5432/backupdbname?sslmode=disable" # Example
```
Comment (samsondav) [inline core/config/v2/docs/secrets.toml:11]:
```suggestion
BackupURL = "postgresql://user:pass@read-replica.example.com:5432/dbname?sslmode=disable" # Example
```

File: core/config/v2/docs/secrets.toml (hunk 3 above)
Comment (samsondav) [reply core/config/v2/docs/secrets.toml:15]: Probably everything in this file should have an ENV var alternative.

**Review by samsondav — APPROVED** [review]

**Review by essamhassan — APPROVED** [review]

**Pull request merged**